cogni-sem/src/main/java/zone/cogni/sem/jena/SparqlUtils.java

```diff
@@ -6,7 +6,8 @@ public enum SparqlUtils {
   ;
 
   public static String escapeLiteral(String literal) {
-    return StringUtils.replace(literal, "'", "\\'");
+    return StringUtils.replace(literal, "'", "\\'")
+                      .replace("\n", "\\n");
```

Comment:
hmm. yeah. there are chars that need to be escaped in general. if it is a normal string and you want to do the normal Java escaping because you need an escaped string. this is probably not the way.

Reply:
`public static String escapeString(String input) {
    return input
            .replace("\\", "\\\\") // Escape backslash
            .replace("\t", "\\t")  // Escape tab
            .replace("\b", "\\b")  // Escape backspace
            .replace("\n", "\\n")  // Escape newline
            .replace("\r", "\\r")  // Escape carriage return
            .replace("\f", "\\f")  // Escape formfeed
            .replace("\'", "\\\'") // Escape single quote
            .replace("\"", "\\\""); // Escape double quote
}
`

is it this you want?
if so, 
`import org.apache.commons.text.StringEscapeUtils;

public static String escapeString(String input) {
    return StringEscapeUtils.escapeJava(input);
}`

Reply:
@natan-cox-cognizone it's the method to escape literals when constructing sparql's
i think the escapeJava stuff will be a bit too much and might alter the final value (or not), so we need to come up with the list of values we want to escape

@christinaspachiou make a unittest for this with a dummy dataset that uses this with insert queries and then just fetch the data via the dataset (model) to see if what we have in the store is what we want

Reply:
ok, thanks for the clarification. 
then we should go to the SPARQL spec or so, no? We are allowed to do both "literal" and 'literal'. So, I am guessing depending on the wrapping using "" or '' we should also escape " or '

Reply:
https://www.w3.org/TR/sparql11-query/#grammarEscapes ?

Reply:
@christinaspachiou check above link for what to escape